common/utils/include/claragenomics/utils/cudautils.hpp

```diff
@@ -89,6 +89,18 @@ __host__ __device__ __forceinline__
     return (value + boundary) & ~(boundary - 1);
 }
 
+template <typename Type>
+void copy(Type* dst, const Type* src, size_t len, cudaStream_t stream)
```

Comment:
I find this a bit confusing: Since the function mimics cudaMemcpy I could expect `len` to be the number of bytes instead of the number of elements.
A simple way out would be to mimic STL's copy instead:
```
template <typename Type>
OutputIterator copy(const Type* src_begin, const Type* src_end, Type* dest_begin, cudaStream_t stream)
```

Reply:
Since we don't have a device_ptr class, I would actually rename it to `device_copy` to avoid name clash with `std::copy`.

Reply:
I noticed you only use this function to get single elements.
How about a
```
template <typename T>
T get_value_from_device(const T* device_ptr, cudaStream_t stream)
```
which implicitly synchronizes the stream, and a
```
template <typename T>
void set_device_value_async(const T* src, T* device_ptr, cudaStream_t stream)
```
which does not?

Reply:
I have added `device_copy_n` instead of `device_copy` for my use case here. 
Feel free to comment on the order of args on all the new methods based on your experience with standards, I ordered them based on what felt natural to my use cases.

Reply:
That's also a good solution. Matches C++'s `copy_n`.